### DISCUSSION TITLE
Best practice for when you need filter list of items in solid

### DISCUSSION THREAD

**Discussion by MadadiSaeed** [solidjs/solid#1170] [Q&A] [answered]
Hi. In my app I have a list of 5000 items and a search box for filtering these items based on the search value and show filtered items instead of the whole items. out of curiosity what is best practice in this scenario. Just a simple example.

**Comment by maxmilton** [comment, marked as answer]
I don't think there's anything I'd consider best practice for this case. Keep it simple and handle things similar to plain JavaScript without a framework (e.g., `Array.prototype.filter()` over your list of items) but leverage Solid's signals + event delegation + control flow components for easy rendering.

Here's a simple playground example: <https://playground.solidjs.com/?hash=-625588573&version=1.4.1>

And real world use of the same kind of thing: <https://github.com/maxmilton/trackx/blob/master/packages/trackx-dash/src/pages/projects/index.tsx#L65-L72>

---

Alternatively, you could use a [custom directive](https://www.solidjs.com/docs/latest/api#use___) to reuse similar logic in multiple places.

**Reply by MadadiSaeed** [reply]
Thanks for the response.
